pkg/ccl/changefeedccl/sink_bench_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2024 The Cockroach Authors.
+
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeedccl
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/url"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/sql/randgen"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/stretchr/testify/require"
+)
+
+func setUpWebHookSinks(b *testing.B) (*cdctest.MockWebhookSink, string) {
+	cert, certEncoded, err := cdctest.NewCACertBase64Encoded()
+	require.NoError(b, err)
+	sinkDest, err := cdctest.StartMockWebhookSink(cert)
+	require.NoError(b, err)
+
+	sinkDestHost, err := url.Parse(sinkDest.URL())
+	require.NoError(b, err)
+
+	params := sinkDestHost.Query()
+	params.Set(changefeedbase.SinkParamCACert, certEncoded)
+	sinkDestHost.RawQuery = params.Encode()
+	return sinkDest, sinkDestHost.String()
+}
+
+func generateRandomizedBytes(rand *rand.Rand) []byte {
+	const tableID = 42
+	dataTypes := []*types.T{types.String, types.Int, types.Decimal, types.Bytes, types.Bool, types.Date, types.Timestamp, types.Float}
+	randType := dataTypes[rand.Intn(len(dataTypes))]
+
+	key, err := keyside.Encode(
+		keys.SystemSQLCodec.TablePrefix(tableID),
+		randgen.RandDatumSimple(rand, randType),
+		encoding.Ascending,
+	)
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
+
+func runWebhookSink(b *testing.B, sinkSrc Sink, rng *rand.Rand, enc Encoder) {
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		bytes := generateRandomizedBytes(rng)
+		ts := hlc.Timestamp{WallTime: rand.Int63() + 1}
```

Comment:
Would use linearly advancing timestamps work here. that is, could this be 

```suggestion
		ts := hlc.Timestamp{WallTime: i}
```

You could also restrict this to the case where you use it below.